src/components/character-controller.js

```diff
@@ -120,9 +123,23 @@ AFRAME.registerComponent("character-controller", {
         z: root.rotation.z * THREE.Math.RAD2DEG
       });
 
-      this.el.setAttribute("position", root.position);
-
       this.pendingSnapRotationMatrix.identity(); // Revert to identity
+
+      //copied from aframe-extras movement-controls
+      if (this.velocity.lengthSq() > EPS) {
+        let start = root.position;
+        end.copy(this.velocity).multiplyScalar(dt / 1000).add(start);
+        const nav = this.el.sceneEl.systems.nav;
+        this.navGroup = this.navGroup || nav.getGroup(start);
+        this.navNode = this.navNode || nav.getNode(start, this.navGroup);
+        this.navNode = nav.clampStep(start, end, this.navGroup, this.navNode, clampedEnd);
+        root.position.copy(clampedEnd);
+        root.position.x += this.velocity.x * dt / 1000;
+        root.position.y += this.velocity.y * dt / 1000;
+        root.z += this.velocity.z * dt / 1000;
```

Comment:
What's this?

Reply:
This whole second applicaiton of velocity looks to be a copy/paste typo https://github.com/donmccurdy/aframe-extras/blob/master/src/controls/movement-controls.js#L124